src/components/body/BodyTableRow.tsx

```diff
@@ -18,19 +25,74 @@ const BodyTableRow: React.FC<IProps> = ({ columns, rowIndex, rowItem }) => {
     [context.bodyRowHeight]
   );
 
+  const { onClick, setEditOptions } = context;
+
+  const customClickHandler: React.MouseEventHandler<HTMLTableDataCellElement> = (
+    evt
+  ) => {
+    evt.preventDefault();
+
+    const value = evt.currentTarget.dataset.value;
+    const colIdx = evt.currentTarget.dataset.col
+    const rowIdx = rowIndex;
+    if(colIdx === undefined)
+      return;
+    onClick?.(evt, value, rowIdx,  Number.parseInt(colIdx));
+    if(setEditOptions)
+      onEditing(evt, Number.parseInt(colIdx), rowIdx);
+  }
+
+  const onEditing = (evt: React.MouseEvent, colIdx: number, rowIdx: number) => {
+    setEditingPosition({ ...editingPosition, col: colIdx, row: rowIdx });
+  };
+
+  const onBlur:React.FocusEventHandler<HTMLInputElement> = (evt) => {
+    const colIdx: number = Number.parseInt(evt.currentTarget.id);
+    const nextState: IDatagridContext = {
+        ...context
+    };
+    const nextData:any = nextState.data;
+    const key= context._colGroup?.[colIdx].key;
+    if (key !== undefined)
+      nextData[rowIndex]["value"][key] = evt.currentTarget.value;
+      setEditingPosition({...setEditingPosition, col: -1, row: -1});
```

Comment:
If you want to update state with previous state, use functional-updates.
ref: https://ko.reactjs.org/docs/hooks-reference.html#functional-updates